js/main.js

```diff
@@ -1,2 +1,70 @@
 /* global data */
 /* exported data */
+
+var $entryForm = document.querySelector('[data-view="entry-form"]');
+var $entryUrl = document.querySelector('#entry-url');
+var $entryTitle = document.querySelector('#entry-title');
+var $entryNotes = document.querySelector('#entry-notes');
+var $entryImage = document.querySelector('.entry-image > img');
+var photoUrl = '';
+var photoTitle = '';
+var notes = '';
+
+if (localStorage.getItem('data')) {
+  for (var key in data) {
+    var previousDataJSON = localStorage.getItem('data');
+    var previousData = JSON.parse(previousDataJSON);
+    data[key] = previousData[key];
+  }
+}
+
+$entryForm.addEventListener('input', function (event) {
+  if (event.target === $entryUrl) {
+    photoUrl = event.target.value;
+  }
+  if (event.target === $entryTitle) {
+    photoTitle = event.target.value;
+  }
+  if (event.target === $entryNotes) {
+    notes = event.target.value;
+  }
+});
+
+$entryUrl.addEventListener('blur', function (event) {
+  if (photoUrl) {
+    $entryImage.setAttribute('src', photoUrl);
+  }
+});
```

Comment:
Instead of using a global variable, you should just check if there is currently a value in the URL `input`, and if so, set the `src` of the image preview.

Reply:
hun, okay, yeah, that simplifies it